### PR TITLE
Add model definition charts to gallery

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### Give a brief description for the solution you have provided

match weights and m u parameters chart

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/charts/m_u_parameters_chart.ipynb [hunk 1]
```diff
@@ -168,15 +165,48 @@
...
+    "\n",
+    "    ![](./img/m_u_parameters_chart_tooltip_2.png)\n",
+    "\n",
+    "    The tooltip of the right chart shows information based on the comparison level bar that the user is hovering over, including:\n",
+    "\n",
+    "    - An explanation of the u probability from the comparison level.\n",
+    "    - The name of the comparison and comparison level.\n",
+    "    - The comparison level condition as an SQL statement.\n",
+    "    - The m and u proability for the comparison level.\n",
+    "    - The resulting bayes factor and match weight for the comparison level."
```
Comment (ThomasHepworth) [inline docs/charts/m_u_parameters_chart.ipynb:197]: Should we label these with subtitles? At the moment it's a tad confusing working out where each image has come from at a glance (this is clarified below, but it's not as obvious when scanning).

I would consider adding some simplistic titles such as:
`Estimated m probability tooltips` and `Estimated u probability tooltips`. 

These suggestions differ from the subtitles already in place on the chart, so you may want to mirror those instead for clarity.

Comment (ThomasHepworth) [reply docs/charts/m_u_parameters_chart.ipynb:197]: Maybe expand on the m/u descriptions? 

Some additional commentary on precisely what the %s mean in practice would be useful for users.

Links to other areas of the docs discussing this would also work.

Comment (ThomasHepworth) [reply docs/charts/m_u_parameters_chart.ipynb:197]: On https://github.com/moj-analytical-services/splink/pull/1551/files#r1317430088, if you feel the description below on "How to interpret the chart" is sufficient, you can ignore this.

File: docs/charts/m_u_parameters_chart.ipynb [hunk 2]
```diff
@@ -185,7 +215,35 @@
    "metadata": {},
    "source": [
     "### Actions to take as a result of the chart\n",
-    "\n"
+    "\n",
+    "As with the `match_weights_chart`, one of the most effective methods to assess a Splink model in to walk through each of the comparison levels of the `m_u_parameters_chart` and sense check the m and u probabilities that have been allocated by the model.\n",
```
Comment (ThomasHepworth) [inline docs/charts/m_u_parameters_chart.ipynb:219]:
```suggestion
    "As with the `match_weights_chart`, one of the most effective methods to assess a Splink model is to walk through each of the comparison levels of the `m_u_parameters_chart` and sense check the m and u probabilities that have been allocated by the model.\n",
```

File: docs/charts/match_weights_chart.ipynb [hunk 3]
```diff
@@ -168,15 +165,46 @@
...
+   "source": [
+    "### How to interpret the chart\n",
+    "\n",
+    "Each bar in the `match_weights_chart` shows the evidence of a match provided by each level in a Splink model (i.e. match weight). As such, the match weight chart provides a summary for the entire Splink model, as it shows the match weights for every type of comparison defined within the model.\n",
+    "\n",
+    "Any Splink score generated to compare two records will add up the evidence (i.e. match weights) for each comparison to come up with a final match weight score, which can then be converted into a probability of a match.\n",
+    "\n",
+    "The first bar chart is the Prior Match Weight, which is the . This can be thought of in the same way as the y-intercept of a simple regression model\n",
+    "\n",
+    "This chart is an aggregation of the [`m_u_parameters_chart`](./m_u_parameters_chart.ipynb). The match weight for a comparison level is simply $log_2(\\frac{m}{u})$."
```
Comment (ThomasHepworth) [inline docs/charts/match_weights_chart.ipynb:207]: Link to Robin's article for more info?

File: docs/charts/match_weights_chart.ipynb [hunk 4]
```diff
@@ -168,15 +165,46 @@
...
    "metadata": {},
    "source": [
-    "### How to interpret the chart\n"
+    "### What the chart shows\n",
+    "\n",
+    "The `match_weights_chart` show the results of a trained Splink model. Each comparison within a model is represented in a bar chart, with a bar showing the evidence for two records being a match (i.e. match weight) for each comparison level.\n",
+    "\n",
+    "??? note \"What the chart tooltip shows\"\n",
+    "\n",
+    "        ![](./img/match_weights_chart_tooltip.png)\n",
```
Comment (ThomasHepworth) [inline docs/charts/match_weights_chart.ipynb:184]: The png is broken on my end

File: docs/charts/match_weights_chart.ipynb [hunk 5]
```diff
@@ -185,7 +213,60 @@
    "metadata": {},
    "source": [
     "### Actions to take as a result of the chart\n",
-    "\n"
+    "\n",
+    "Some heuristics to help assess Splink models with the `match_weights_chart`:\n",
+    "\n",
+    "#### Reducing match weights within a comparison\n",
```
Comment (ThomasHepworth) [inline docs/charts/match_weights_chart.ipynb:219]: Maybe reword this subtitle? It makes it sound as if users should reduce the number of comparison levels in their comparison.

Contracting comparison match weights
or even just
Reducing comparison match weights

File: docs/charts/match_weights_chart.ipynb (hunk 3 above)
Comment (RossKen) [reply docs/charts/match_weights_chart.ipynb:207]: is there a specific article you are thinking of here?

File: docs/charts/m_u_parameters_chart.ipynb (hunk 1 above)
Comment (RossKen) [reply docs/charts/m_u_parameters_chart.ipynb:197]: Have added a link to the fellegi sunter docs

File: docs/charts/match_weights_chart.ipynb (hunk 4 above)
Comment (RossKen) [reply docs/charts/match_weights_chart.ipynb:184]: fixed

File: docs/charts/match_weights_chart.ipynb (hunk 5 above)
Comment (RossKen) [reply docs/charts/match_weights_chart.ipynb:219]: have added alternate subtitle

**Review by ThomasHepworth — APPROVED** [review]
Thanks Ross, looks good
